src/utils.php

```diff
@@ -189,3 +171,8 @@ function getVendorDir(\stdClass $composerJson = null): string
 {
     return $composerJson->config->{'vendor-dir'} ?? 'vendor';
 }
+
+function strStartsWith(string $haystack, string $prefix): bool
+{
+    return empty($prefix) || strpos($haystack, $prefix) === 0;
```

Comment:
Is this function really needed? `substr($haystack, 0, strlen($prefix)) === $prefix` should be faster (depending on heystack size) because it doesn't have to search the whole string, just check the first `strlen($prefix)` characters for identity.  In a loop you only have to get the prefix length once. And it also returns `true` for an empty prefix, since `substr($haystack, 0, 0)` is the empty string.